vortex-ipc/src/messages.rs

```diff
@@ -186,15 +192,120 @@ impl<'a> WriteFlatBuffer for IPCArray<'a> {
+    array: &'_ ArrayData,
+) -> WIPOffset<ArrayStats<'a>> {
+    let primitive_ptype = match array.dtype() {
+        DType::Primitive(ptype, _) => Some(ptype),
+        _ => None,
+    };
+
+    let min = primitive_ptype.and_then(|ptype| {
+        match_each_native_ptype!(ptype, |$T| {
+            array.statistics().compute_min::<$T>().ok().map(|min| {
+                Primitive(PrimitiveScalar::some(min)).write_flatbuffer(fbb)
+            })
+        })
+    });
+    let max = primitive_ptype.and_then(|ptype| {
+        match_each_native_ptype!(ptype, |$T| {
+            array.statistics().compute_max::<$T>().ok().map(|max| {
+                Primitive(PrimitiveScalar::some(max)).write_flatbuffer(fbb)
+            })
+        })
+    });
+
+    let is_constant = array
+        .statistics()
+        .compute_is_constant()
+        .ok()
+        .map(|v| Bool(BoolScalar::some(v)).write_flatbuffer(fbb));
+    let is_sorted = array
+        .statistics()
+        .compute_is_sorted()
+        .ok()
+        .map(|v| Bool(BoolScalar::some(v)).write_flatbuffer(fbb));
+    let is_strict_sorted = array
+        .statistics()
+        .compute_is_strict_sorted()
+        .ok()
+        .map(|v| Bool(BoolScalar::some(v)).write_flatbuffer(fbb));
+
+    let run_count = array
+        .statistics()
+        .compute_run_count()
+        .ok()
+        .map(|v| Primitive(PrimitiveScalar::some(v as u64)).write_flatbuffer(fbb));
+    let true_count = array
+        .statistics()
+        .compute_true_count()
+        .ok()
+        .map(|v| Primitive(PrimitiveScalar::some(v as u64)).write_flatbuffer(fbb));
+    let null_count = array
+        .statistics()
+        .compute_null_count()
+        .ok()
+        .map(|v| Primitive(PrimitiveScalar::some(v as u64)).write_flatbuffer(fbb));
+
+    let bit_width_freq = array
+        .statistics()
+        .compute_bit_width_freq()
+        .ok()
+        .map(|v| {
+            v.iter()
+                .map(|&inner| inner as u64)
+                .map(PrimitiveScalar::some)
+                .map(Primitive)
+                .map(|v| v.write_flatbuffer(fbb))
+                .collect_vec()
+        })
+        .map(|v| fbb.create_vector(v.as_slice()));
+
+    let trailing_zero_freq = array
+        .statistics()
+        .compute_trailing_zero_freq()
+        .ok()
+        .map(|v| {
+            v.iter()
+                .map(|&inner| inner as u64)
+                .map(PrimitiveScalar::some)
+                .map(Primitive)
+                .map(|v| v.write_flatbuffer(fbb))
+                .collect_vec()
+        })
+        .map(|v| fbb.create_vector(v.as_slice()));
+
+    let stat_args = &ArrayStatsArgs {
+        min,
+        max,
+        is_sorted,
+        is_strict_sorted,
+        is_constant,
+        run_count,
+        true_count,
+        null_count,
+        bit_width_freq,
+        trailing_zero_freq,
+    };
+
+    let stats = ArrayStats::create(fbb, stat_args);
```

Comment:
I pet peeve of mine is useless variables like this 😅

Reply:
Writing this on the back of my hand